src/core/stdcpp/string.d

```diff
@@ -167,6 +167,24 @@ extern(D):
     ///
     inout(T)[] opIndex() inout pure nothrow @safe @nogc                     { return as_array(); }
 
+    /// Two `basic_string`s are equal if they represent the same sequence of code units.
+    bool opEquals(scope const ref basic_string s) const pure nothrow @safe  { return as_array == s.as_array; }
+    /// ditto
+    bool opEquals(basic_string s) const                                     { return as_array == rhs.as_array; }
```

Comment:
What is this overload for? The function above is the same. Also, the operand on the right is `rhs.as_array`, but the parameter is `s`...?

Reply:
`basic_string` has a destructor that zeroes its member so passing it by reference does less work and when passing by value one can't convert non-const to const.

Reply:
Also, oops fixed.

Reply:
Why would you ever pass by value though? I think this by-val comparison shouldn't exist.

Reply:
You may be right. A pass-by-value overload is necessary to make things like `s == some_function_that_returns_a_string()` work when not compiling with `-preview=rvaluerefparam`, but when `-preview=rvaluerefparam` is enabled then an explicit pass-by-value overload is a pessimization.

Reply:
`-preview=rvaluerefparam` has been in testing for a long time now; hopefully it becomes enabled by default in the near-ish future.

Reply:
Removed the pass-by-value overloads. This may cause existing code in the wild using stdcpp.string with default struct equality to stop compiling, but as that code was only producing the correct result for small strings this may be acceptable.

Reply:
I'm fine with this. Thanks.